components/datetime/src/pattern/item/ule.rs

```diff
@@ -0,0 +1,95 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use super::PatternItem;
+use crate::fields;
+use core::convert::TryFrom;
+use zerovec::ule::{AsULE, EqULE, ULE};
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct PatternItemULE(pub [u8; 3]);
+
+impl PatternItemULE {
+    pub fn is_literal_from_u8(byte: u8) -> bool {
+        byte & 0b1000_0000 != 0
+    }
+
+    pub fn clear_type_in_u8(byte: u8) -> u8 {
+        byte ^ 0b1000_0000
+    }
+
+    pub fn set_literal_in_u8(byte: u8) -> u8 {
+        byte | 0b1000_0000
+    }
+
+    pub fn bytes_in_range(value: (&u8, &u8, &u8)) -> bool {
+        match Self::is_literal_from_u8(*value.0) {
+            false => fields::Field::bytes_in_range(value.1, value.2),
+            true => {
+                let u = u32::from_le_bytes([
+                    *value.2,
+                    *value.1,
+                    Self::clear_type_in_u8(*value.0),
```

Comment:
Suggestion: if you switch the polarity of the `type` bit (1 means "field" rather than "literal"), then you can pass `*value.0` in here directly, and it doesn't affect what happens for the non-literal case.